src/Raven.Studio/typescript/viewmodels/database/tasks/editOlapEtlTask.ts

```diff
@@ -0,0 +1,583 @@
+    
+    editedOlapEtl = ko.observable<ongoingTaskOlapEtlEditModel>();
+    isAddingNewOlapEtlTask = ko.observable<boolean>(true);
+    
+    transformationScriptSelectedForEdit = ko.observable<ongoingTaskOlapEtlTransformationModel>();
+    editedTransformationScriptSandbox = ko.observable<ongoingTaskOlapEtlTransformationModel>();
+
+    olapTableSelectedForEdit = ko.observable<ongoingTaskOlapEtlTableModel>();
+    editedOlapTableSandbox = ko.observable<ongoingTaskOlapEtlTableModel>();
+
+    possibleMentors = ko.observableArray<string>([]);
+    olapEtlConnectionStringsNames = ko.observableArray<string>([]);
+
+    connectionStringDefined: KnockoutComputed<boolean>;
+    testConnectionResult = ko.observable<Raven.Server.Web.System.NodeConnectionTestResult>();
+    
+    spinners = {
+        test: ko.observable<boolean>(false),
+        save: ko.observable<boolean>(false)
+    };
+    
+    fullErrorDetailsVisible = ko.observable<boolean>(false);
+    shortErrorText: KnockoutObservable<string>;
+    
+    collectionNames: KnockoutComputed<string[]>;
+    
+    showAdvancedOptions = ko.observable<boolean>(false);
+    showEditTransformationArea: KnockoutComputed<boolean>;
+    showEditOlapTableArea: KnockoutComputed<boolean>;
+
+    createNewConnectionString = ko.observable<boolean>(false);
+    newConnectionString = ko.observable<connectionStringOlapEtlModel>();
+
+    serverConfiguration = ko.observable<periodicBackupServerLimitsResponse>(); // needed for olap local destination in connection string
+
+    constructor() {
+        super();
+        this.bindToCurrentInstance("useConnectionString",
+                                   "useCollection",
+                                   "testCredentials",
+                                   "removeTransformationScript",
+                                   "cancelEditedTransformation",
+                                   "cancelEditedOlapTable",
+                                   "saveEditedTransformation",
+                                   "saveEditedOlapTable",
+                                   "syntaxHelp",
+                                   "toggleAdvancedArea",
+                                   "deleteOlapTable",
+                                   "editOlapTable");
+
+        aceEditorBindingHandler.install();
+    }
+
+    activate(args: any) {
+        super.activate(args);
+        const deferred = $.Deferred<void>();
+
+        if (args.taskId) {
+            // 1. Editing an Existing task
+            this.isAddingNewOlapEtlTask(false);
+
+            getOngoingTaskInfoCommand.forOlapEtl(this.activeDatabase(), args.taskId)
+                .execute()
+                .done((result: Raven.Client.Documents.Operations.OngoingTasks.OngoingTaskOlapEtlDetails) => {
+                    this.editedOlapEtl(new ongoingTaskOlapEtlEditModel(result));
+                    this.showAdvancedOptions(!!result.Configuration.MaxNumberOfItemsInRowGroup);
```

Comment:
let's move that logic to separate method - it will be helpful if we add extra conditions in future.

Reply:
removed